doc/api/assert.md

```diff
@@ -632,14 +632,16 @@ changes:
 > Stability: 0 - Deprecated: Use `assert.fail([message])` or other assert
 > functions instead.
 
-If `message` is falsy, the error message is set as the values of `actual` and
-`expected` separated by the provided `operator`. If just the two `actual` and
-`expected` arguments are provided, `operator` will default to `'!='`. If
-`message` is provided as third argument it will be used as the error message and
-the other arguments will be stored as properties on the thrown object. If
-`stackStartFn` is provided, all stack frames above that function will be
-removed from stacktrace (see [`Error.captureStackTrace`]). If no arguments are
-given, the default message `Failed` will be used.
+If just the two `actual` and `expected` arguments are provided, `operator` will
+default to `'!='`. If `message` is provided as third argument it will be used
+as the error message and the other arguments will be stored as properties on
+the thrown object. If `stackStartFn` is provided, all stack frames above that
+function will be removed from stacktrace (see [`Error.captureStackTrace`]).
+If no arguments are given, the default message `Failed` will be used.
+
+If `message` is provided but falsy, and no other optional arguments are
+specified, the error message is set as the values of `actual` and `expected`
+separated by the provided `operator`(`'fail'`).
```

Comment:
I suggest to change this to the following (I tried to be very precise about the actual expected behavior):

```
If `message` is falsy, the error message is set as the values of `actual` and
`expected` separated by the provided `operator`. If `operator` is falsy and no
message is set, it'll default to `'!='`. If `message` is truthy the `operator`
will default to `'fail'` and `message` will be used as error message. Other
arguments will be stored as properties on the thrown object. If `stackStartFn`
is provided, all stack frames above that function will be removed from
stacktrace (see [`Error.captureStackTrace`]).
```

Reply:
Very precise and concise.